plugins/GSdx/GSDump.cpp

```diff
@@ -22,74 +22,165 @@
 #include "stdafx.h"
 #include "GSDump.h"
 
-GSDump::GSDump()
-	: m_gs(NULL)
-	, m_frames(0)
-	, m_extra_frames(0)
+GSDumpBase::GSDumpBase(const string& fn)
+	: m_frames(0)
+	, m_extra_frames(2)
 {
+	m_gs = fopen(fn.c_str(), "wb");
 }
 
-GSDump::~GSDump()
+GSDumpBase::~GSDumpBase()
 {
-	Close();
+	if(m_gs)
+		fclose(m_gs);
 }
 
-void GSDump::Open(const string& fn, uint32 crc, const GSFreezeData& fd, const GSPrivRegSet* regs)
+void GSDumpBase::AddHeader(uint32 crc, const GSFreezeData& fd, const GSPrivRegSet* regs)
 {
-	m_gs = fopen((fn + ".gs").c_str(), "wb");
+	AppendRawData(&crc, 4);
+	AppendRawData(&fd.size, 4);
+	AppendRawData(fd.data, fd.size);
+	AppendRawData(regs, sizeof(*regs));
+}
 
-	m_frames = 0;
-	m_extra_frames = 2;
+void GSDumpBase::Transfer(int index, const uint8* mem, size_t size)
+{
+	if (size == 0)
+		return;
 
-	if(m_gs)
-	{
-		fwrite(&crc, 4, 1, m_gs);
-		fwrite(&fd.size, 4, 1, m_gs);
-		fwrite(fd.data, fd.size, 1, m_gs);
-		fwrite(regs, sizeof(*regs), 1, m_gs);
-	}
+	AppendRawData(0);
+	AppendRawData(index);
+	AppendRawData(&size, 4);
+	AppendRawData(mem, size);
 }
 
-void GSDump::Close()
+void GSDumpBase::ReadFIFO(uint32 size)
 {
-	if(m_gs) {fclose(m_gs); m_gs = NULL;}
+	if (size == 0)
+		return;
+
+	AppendRawData(2);
+	AppendRawData(&size, 4);
 }
 
-void GSDump::Transfer(int index, const uint8* mem, size_t size)
+bool GSDumpBase::VSync(int field, bool last, const GSPrivRegSet* regs)
 {
-	if(m_gs && size > 0)
-	{
-		fputc(0, m_gs);
-		fputc(index, m_gs);
-		fwrite(&size, 4, 1, m_gs);
-		fwrite(mem, size, 1, m_gs);
-	}
+	AppendRawData(3);
+	AppendRawData(regs, sizeof(*regs));
+
+	AppendRawData(1);
+	AppendRawData(field);
+
+	if (last)
+		m_extra_frames--;
+
+	return ((++m_frames & 1) == 0 && last && (m_extra_frames < 0));
 }
 
-void GSDump::ReadFIFO(uint32 size)
+//////////////////////////////////////////////////////////////////////
+// GSDump implementation
+//////////////////////////////////////////////////////////////////////
+
+GSDump::GSDump(const string& fn, uint32 crc, const GSFreezeData& fd, const GSPrivRegSet* regs) : GSDumpBase(fn + ".gs")
```

Comment:
Maybe put the base class constructor on a new line? IMO it's a bit hard to notice. Ditto GSDumpXz.